attestation-agent/attester/src/snp/mod.rs

```diff
@@ -47,4 +49,18 @@ impl Attester for SnpAttester {
 
         serde_json::to_string(&evidence).context("Serialize SNP evidence failed")
     }
+
+    async fn check_init_data(&self, init_data: &[u8]) -> Result<()> {
```

Comment:
I think of an issue: I'm not sure whether it's a good idea to pass init data as an API parameter, because this will lead to any user mode program can pass in an init data for AA to check, which may lead to violent collision cracking of initdata.

Perhaps a better way is to let AA read the init data to be verified from a file with a fixed path. This can ensure that only users with specific permissions (which can modify that file) can use this function of AA by setting file access permissions.

Reply:
I might get your point. The core idea of initdata is to protect the integrity, while the confidentiality is not.

Because the init data plaintext is set by untrusted host when the VM is created, we might not need to worry about the confidentiality.

Furthermore, the API of AA is now provided by forwarding from ASR(API Server Restful) via restful api, which could be configured which api of AA could be exposed to app containers. This could help to prevent unexpected access.

Reply:
Interesting point. I think as @Xynnn007 says it is probably not an issue. The InitData is not secret and this API won't be available to the workload containers. I can't think of any attack where a userspace component could leverage this API to do something dangerous.